spec/liquid_tags/gist_tag_spec.rb

```diff
@@ -71,6 +71,12 @@ def generate_script(link, option = "")
       end.to raise_error(StandardError)
     end
 
+    it "rejects empty gist url" do
+      expect do
+        generate_new_liquid
```

Comment:
This test totally works but I had to open up the rest of this file to figure out how this method created an empty gist URL since you can't really tell based on the method name. One way to improve this might be to use a keyword for the method and clean up the rest of the file 
```ruby
    def generate_new_liquid(link:)
      Liquid::Template.register_tag("gist", GistTag)
      Liquid::Template.parse("{% gist #{link} %}")
    end
```

A comment might also be helpful. Not really a blocker but figured I would point it out.

Reply:
Thank you @mstruve.  I did try to convert link to keyword as above

```ruby
    def generate_new_liquid(link:)
      Liquid::Template.register_tag("gist", GistTag)
      Liquid::Template.parse("{% gist #{link} %}")
    end
```

but it was causing some tests to fail

```
$ bundle exec rspec spec/liquid_tags/gist_tag_spec.rb
[Zonebie] Setting timezone: ZONEBIE_TZ="Monrovia"
FFFF...

Failures:

  1) GistTag#link accepts proper gist url
     Failure/Error:
       def generate_new_liquid(link:)
         Liquid::Template.register_tag("gist", GistTag)
         Liquid::Template.parse("{% gist #{link} %}")
       end
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: link)
     # ./spec/liquid_tags/gist_tag_spec.rb:28:in `generate_new_liquid'
     # ./spec/liquid_tags/gist_tag_spec.rb:45:in `block (4 levels) in <top (required)>'
     # ./spec/liquid_tags/gist_tag_spec.rb:44:in `each'
     # ./spec/liquid_tags/gist_tag_spec.rb:44:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (2 levels) in <top (required)>'

  2) GistTag#link handles 'file' option
     Failure/Error:
       def generate_new_liquid(link:)
         Liquid::Template.register_tag("gist", GistTag)
         Liquid::Template.parse("{% gist #{link} %}")
       end
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: link)
     # ./spec/liquid_tags/gist_tag_spec.rb:28:in `generate_new_liquid'
     # ./spec/liquid_tags/gist_tag_spec.rb:51:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (2 levels) in <top (required)>'

  3) GistTag#link allows embed of specific version
     Failure/Error:
       def generate_new_liquid(link:)
         Liquid::Template.register_tag("gist", GistTag)
         Liquid::Template.parse("{% gist #{link} %}")
       end
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: link)
     # ./spec/liquid_tags/gist_tag_spec.rb:28:in `generate_new_liquid'
     # ./spec/liquid_tags/gist_tag_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (2 levels) in <top (required)>'

  4) GistTag#link allows embed of specific version with 'file' option
     Failure/Error:
       def generate_new_liquid(link:)
         Liquid::Template.register_tag("gist", GistTag)
         Liquid::Template.parse("{% gist #{link} %}")
       end
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: link)
     # ./spec/liquid_tags/gist_tag_spec.rb:28:in `generate_new_liquid'
     # ./spec/liquid_tags/gist_tag_spec.rb:63:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:130:in `block (2 levels) in <top (required)>'

Finished in 2.54 seconds (files took 5.2 seconds to load)
7 examples, 4 failures

Failed examples:

rspec ./spec/liquid_tags/gist_tag_spec.rb:43 # GistTag#link accepts proper gist url
rspec ./spec/liquid_tags/gist_tag_spec.rb:50 # GistTag#link handles 'file' option
rspec ./spec/liquid_tags/gist_tag_spec.rb:56 # GistTag#link allows embed of specific version
rspec ./spec/liquid_tags/gist_tag_spec.rb:61 # GistTag#link allows embed of specific version with 'file' option

Coverage report generated for RSpec to /home/rafi/Projects/dev.to/coverage. 1345 / 22585 LOC (5.96%) covered.
SimpleCov failed with exit 1%   
```
so I did not commit those changes

Reply:
I could change those tests to accommodate this change but I was not sure if I should.